### PR TITLE
Compare properties as JSON instead of strings

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Currently, we read all values from `properties` from Postgres as text and compare that. This PR does not converts the `properties` to text but instead converts the `parameter`s to JSON and compare that instead

## 🔗 Related links

- [Make sure, that `"2.0"` is not the same as `2.0` when querying](https://app.asana.com/0/1203543021352041/1203763211971484/f) _(internal)_
- [Support querying for non-existing fields](https://app.asana.com/0/1203543021352041/1203763211971481/f) _(internal)_

## 🔍 What does this change?

- Remove unused `Timestamp` parameter type
- Creates a (hidden) parameter type `Any`, which can hold any JSON value
- Adds proper conversion for the `Any` field
- Extends the possible values of `JsonField`, this also turns the `Text` field into a `StaticText` fields to ensure, that no dynamic strings can hit the database

## ❓ How to test this?

This can be tested with the REST endpoint tests:
1. Run the full test suite once to populate the database
2. Insert a person entity with a name of `"2.0"`, one with `2.0` (we currently don't have entity validation, so for the sake of testing it's the easiest way), and one with `null`.
3. Query for the name property with different parameters (`"2.0"`, `2.0`, `null` is currently not supported, make sure `null` fails), e.g. `{ parameter: "2.0" }`
4. Replace `parameter` with `null` and make sure it's actually testing for existence

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs [hunk 1]
```diff
@@ -102,17 +103,22 @@ macro_rules! impl_ontology_column {
...
+                                }
+                                JsonField::JsonParameter(index) => {
+                                    write!(fmt, r#"."schema"->${index}"#)
+                                }
+                                JsonField::StaticText(field) => {
+                                    write!(fmt, r#"."schema"->>'{field}'"#)
+                                }
+                                JsonField::StaticJson(field) => {
+                                    write!(fmt, r#"."schema"->'{field}'"#)
+                                }
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs:119]: The above is a leading question for these statements specifically, would be worried of any sort of sql injection. I suppose we don't implement the access syntax and we shouldn't have any issues yet, but just want to make sure this is in mind

**Review by thehabbos007 — COMMENTED** [review]
Just have a couple comments for now. I do see it would be nice to have a builder syntax for the filters

File: packages/graph/hash_graph/lib/graph/src/store/query/filter.rs [hunk 2]
```diff
@@ -312,9 +309,9 @@ impl fmt::Display for ParameterConversionError {
             Parameter::Boolean(boolean) => boolean.to_string(),
             Parameter::Number(number) => number.to_string(),
             Parameter::Text(text) => text.to_string(),
+            Parameter::Any(_) => "JSON value".to_owned(),
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/query/filter.rs:312]: I worry that the error message could get pretty big. I could match on the actual values and decide what to print. What do you think?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs [hunk 3]
```diff
@@ -49,8 +49,9 @@ impl Transpile for Table {
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub enum JsonField<'p> {
     Json(&'p Cow<'p, str>),
-    Text(&'p Cow<'p, str>),
-    Parameter(usize),
+    JsonParameter(usize),
+    StaticText(&'static str),
+    StaticJson(&'static str),
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs:54]: Yes, exactly. We did this previously, but having it a `'static str` is more safe.

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs:119]: I noticed that as well. Maybe we want to `panic!` on `JsonField::Json`?

File: packages/graph/hash_graph/lib/graph/src/store/query/filter.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/query/filter.rs:312]: That's a fair concern, i see that we would likely not want that. I think this is fine then compared to trying to select thing in a smart way

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs (hunk 1 above)
Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs:119]: Yes, I think that would be nice :)

File: packages/graph/hash_graph/lib/graph/src/store/query/filter.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/query/filter.rs]: I added a better output in 0993f3a8c932c9a5cdb5cb2188ff2e4e61bc73d2

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/table.rs]: We are now `panic!`ing 8abaa19018691a7c84e29f45760f7b9f16cdb823

File: packages/graph/hash_graph/lib/graph/src/store/query/filter.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/query/filter.rs]: Nice, this is better!

**Review by thehabbos007 — DISMISSED** [review]
Looks good to me! Tested locally and worked as expected (the "2.0" vs 2.0 and null not working still)
Thanks!

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**